packages/common/src/services/wallet-client/WalletClient.ts

```diff
@@ -62,10 +62,35 @@ export class WalletClient {
   }
 
   async transferTokensFromEthToSol(): Promise<void> {
-    const balance = await this.audiusBackendInstance.getBalance(true)
-    if (!isNullOrUndefined(balance) && balance.gt(new BN('0'))) {
-      await this.audiusBackendInstance.transferAudioToWAudio(balance)
+    const initialWAudioBalance = await this.getCurrentWAudioBalance()
+    const ercAudioBalance = await this.audiusBackendInstance.getBalance(true)
+    if (
+      !isNullOrUndefined(ercAudioBalance) &&
+      ercAudioBalance.gt(new BN('0'))
+    ) {
+      await this.audiusBackendInstance.transferAudioToWAudio(ercAudioBalance)
     }
+
+    // At this point we know we have transfered ERC $AUDIO into the wormhole,
+    // but we don't know if we've received SPL $AUDIO back yet, so poll
+    await new Promise((resolve, reject) => {
```

Comment:
We actually have a fn for this! `pollForBalanceChange` in `solana.ts` here: https://github.com/AudiusProject/audius-protocol/blob/3fa6a97f057ff2f4e4fa90d0f7be4de05f680c3a/packages/common/src/services/audius-backend/solana.ts#L250

Reply:
noice